modules/base/rendering/pointcloud/renderablepointcloud.cpp

```diff
@@ -0,0 +1,1020 @@
+        glVertexAttribPointer(
+            colorParamAttrib,
+            1,
+            GL_FLOAT,
+            GL_FALSE,
+            attibutesPerPoint * sizeof(float),
+            reinterpret_cast<void*>(attributeOffset * sizeof(float))
+        );
+        attributeOffset += 1;
+    }
+
+    if (_hasDatavarSize) {
+        GLint scalingAttrib = _program->attributeLocation("in_scalingParameter");
+        glEnableVertexAttribArray(scalingAttrib);
+        glVertexAttribPointer(
+            scalingAttrib,
+            1,
+            GL_FLOAT,
+            GL_FALSE,
+            attibutesPerPoint * sizeof(float),
+            reinterpret_cast<void*>(attributeOffset * sizeof(float))
+        );
+        attributeOffset += 1;
+    }
+
+    glBindVertexArray(0);
+
+    _dataIsDirty = false;
+}
+
+void RenderablePointCloud::updateSpriteTexture() {
+    bool shouldUpdate = _hasSpriteTexture && _spriteTextureIsDirty &&
+        !_spriteTexturePath.value().empty();
+
+    if (!shouldUpdate) {
+        return;
+    }
+
+    ZoneScopedN("Sprite texture");
+    TracyGpuZone("Sprite texture");
+
+    ghoul::opengl::Texture* texture = _spriteTexture;
+
+    unsigned int hash = ghoul::hashCRC32File(_spriteTexturePath);
+
+    _spriteTexture = BaseModule::TextureManager.request(
+        std::to_string(hash),
+        [path = _spriteTexturePath]() -> std::unique_ptr<ghoul::opengl::Texture> {
+            std::filesystem::path p = absPath(path);
+            LINFO(fmt::format("Loaded texture from {}", p));
+            std::unique_ptr<ghoul::opengl::Texture> t =
+                ghoul::io::TextureReader::ref().loadTexture(p.string(), 2);
+            t->uploadTexture();
+            t->setFilter(ghoul::opengl::Texture::FilterMode::AnisotropicMipMap);
+            t->purgeFromRAM();
+            return t;
+        }
+    );
+
+    BaseModule::TextureManager.release(texture);
+    _spriteTextureIsDirty = false;
+}
+
+int RenderablePointCloud::currentColorParameterIndex() const {
+    const properties::OptionProperty& property =
+        _colorSettings.colorMapping->dataColumn;
+
+    if (!_hasColorMapFile || property.options().empty()) {
+        return 0;
+    }
+
+    return _dataset.index(property.option().description);
+}
+
+int RenderablePointCloud::currentSizeParameterIndex() const {
+    const properties::OptionProperty& property =
+        _sizeSettings.sizeMapping.parameterOption;
+
+    if (!_hasDatavarSize || property.options().empty()) {
+        return 0;
+    }
+
+    return _dataset.index(property.option().description);
+}
+
+std::vector<float> RenderablePointCloud::createDataSlice() {
+    ZoneScoped;
+
+    if (_dataset.entries.empty()) {
+        return std::vector<float>();
+    }
+
+    std::vector<float> result;
+    result.reserve(nAttributesPerPoint() * _dataset.entries.size());
+
+    // what datavar in use for the index color
+    int colorParamIndex = currentColorParameterIndex();
```

Comment:
Consider updating rendering code so that we only need to update these values when the color parameter option changes, instead of regenerating the entire data slice

(Same for the size variable below)